src/index.js

```diff
@@ -1,21 +1,55 @@
-import ReactDOM from "react-dom";
-import React from "react";
+/**
+ * External dependencies
+ */
+import ReactDOM from 'react-dom';
+import React from 'react';
+import head from 'lodash/head';
 import { DropZoneProvider, DropZone } from '@wordpress/components';
 import { withState } from '@wordpress/compose';
+import apiFetch from '@wordpress/api-fetch';
+
+const validFileTypes = [ 'text/xml' ];
+const isValidFileType = type => validFileTypes.includes( type );
 
 const MyDropZone = withState( {
 	hasDropped: false,
-} )( ( { hasDropped, setState } ) => (
-	<DropZoneProvider>
-		<div>
-			{ hasDropped ? 'Dropped!' : 'Drop something here' }
-			<DropZone 
-				onFilesDrop={ () => setState( { hasDropped: true } ) }
-				onHTMLDrop={ () => setState( { hasDropped: true } )  }
-				onDrop={ () => setState( { hasDropped: true } ) } 
-			/>
-		</div>
-	</DropZoneProvider>
-) );
+} )( ( { hasDropped, setState } ) => {
+	const doIt = async ( files = [] ) => {
+		const file = head( files );
+		const { type, size } = file;
+
+		if ( ! size ) {
+			console.error( 'Cannot upload an empty file' );
+			return;
+		}
+
+		if ( ! isValidFileType( type ) ) {
+			console.error( `File type ${ type } is not supported` );
+			return;
+		}
+
+		setState( { hasDropped: true } );
+
+		const body = new FormData();
+		body.append( 'import', file );
+		body.append( 'status', 'private' );
+
+		apiFetch( {
+			method: 'POST',
+			path: '/wp/v2/wp-file-imports',
```

Comment:
Should we add a @todo here to dynamically generate the path? Realize this is still POC-realm, but just in-case we copy this on and forget.

Reply:
I think once we start varying the behavior it will be apparent the should be the result of a function call.